src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -6242,15 +6242,26 @@ private ScanTypeFlags ScanType(ParseTypeMode mode, out SyntaxToken lastTokenOfTy
                 }
             }
 
-            if (this.CurrentToken.Kind == SyntaxKind.IdentifierToken)
+            // Handle :: as well for error case of an alias used without a preceding identifier.
+            if (this.CurrentToken.Kind is SyntaxKind.IdentifierToken or SyntaxKind.ColonColonToken)
             {
-                result = this.ScanNamedTypePart(out lastTokenOfType);
-                if (result == ScanTypeFlags.NotType)
+                bool isAlias;
+                if (this.CurrentToken.Kind is SyntaxKind.ColonColonToken)
                 {
-                    return ScanTypeFlags.NotType;
+                    lastTokenOfType = null;
+                    result = ScanTypeFlags.NonGenericTypeOrExpression;
+                    isAlias = true;
                 }
+                else
+                {
+                    result = this.ScanNamedTypePart(out lastTokenOfType);
+                    if (result == ScanTypeFlags.NotType)
+                    {
+                        return ScanTypeFlags.NotType;
+                    }
 
-                bool isAlias = this.CurrentToken.Kind == SyntaxKind.ColonColonToken;
+                    isAlias = this.CurrentToken.Kind == SyntaxKind.ColonColonToken;
+                }
```

Comment:
the above is so that `::A.B<C> x;` is parsed by a type followed by a name, not as two binary expressions (`<` and `>`).